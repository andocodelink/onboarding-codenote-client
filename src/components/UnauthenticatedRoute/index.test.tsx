import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount, render } from 'enzyme';
import { Button } from 'react-bootstrap';

import { UnauthenticatedRoute, mapStateToProps, querystring } from './index';

describe('Home screen', () => {
  const userHasAuthenticated = () => ({});
  const jdomAlert = window.alert;

  beforeAll(() => {
    window.alert = jest.fn();
  });

  afterAll(() => {
    window.alert = jdomAlert;
  })
  
  test('renders with authentication', () => {
    const wrapper = shallow(<UnauthenticatedRoute id='id1' isAuthenticated={true} component={Button as any} props={{id: 'childId1'}}/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders without authentication', () => {
    const wrapper = shallow(<UnauthenticatedRoute id='id1' isAuthenticated={false} component={Button as any} props={{id: 'childId1'}}/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('map state to props', () => {
    let props = mapStateToProps({
      authenticate: {
        isAuthenticated: true 
      }
    });
    expect(props.isAuthenticated).toBe(true);
  });

  test('query string', () => {
    let actual = querystring("redirect", "http://facebook.com?redirect=codelink.io");
    expect(actual).toEqual("codelink.io");
  });

})