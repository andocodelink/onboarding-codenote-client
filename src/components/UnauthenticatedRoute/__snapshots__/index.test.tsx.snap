// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen renders with authentication 1`] = `
<Route
  id="id1"
  props={
    Object {
      "id": "childId1",
    }
  }
  render={[Function]}
/>
`;

exports[`Home screen renders without authentication 1`] = `
<Route
  id="id1"
  props={
    Object {
      "id": "childId1",
    }
  }
  render={[Function]}
/>
`;
